src/lib/http-agent.js

```diff
@@ -85,7 +83,25 @@ const getAgent = async ({ httpProxy, certificateFile, exit }) => {
   }
 
   const agent = new HttpsProxyAgentWithCA(opts)
-  return agent
+  response = { ...response, agent }
+  return response
+}
+
+const getAgent = async ({ httpProxy, certificateFile }) => {
+  const result = await tryGetAgent({ httpProxy, certificateFile })
+  if (!result) {
+    return
+  }
+  if (result.error) {
+    log(NETLIFYDEVERR, result.error, result.message || '')
+    exit(1)
+  }
+  if (result.warning) {
+    log(NETLIFYDEVWARN, result.warning, result.message || '')
+  }
+  if (result.agent) {
```

Comment:
[pebble] Since `result.agent` is either `undefined` or an `Agent` instance, this line might be removed, i.e. we might be able to always `result.agent`, regardless of whether it is defined or not.

Reply:
Yep. Changed this post destructuring.